src/components/reusable/stepper/stepper.ts

```diff
@@ -0,0 +1,127 @@
+import { LitElement, html } from 'lit';
+import {
+  customElement,
+  property,
+  queryAssignedElements,
+} from 'lit/decorators.js';
+
+import stepperStyles from './stepper.scss';
+import './stepperItem';
+
+/**
+ * Stepper
+ * @slot unnamed - Slot for step items.
+ * @fires on-click - Captures the event and emits the active step and original event details when click on any step title. This is only for procedure type stepper. Status stepper doesn't emit this event.
+ */
+
+@customElement('kyn-stepper')
+export class Stepper extends LitElement {
+  static override styles = stepperStyles;
+
+  /** Stepper type `'procedure'` & `'status'`.
+   *
+   * procedure: Allows a user to move through a series of steps that need to be completed, such as filling out a series of forms. The user can therefore know where they are in the sequence, and can go back to previous steps if needed. Procedure should use the `'large'` size stepper.
+   *
+   * status: Should not allow the user navigate to previous steps for ex: sequential forms, payment gateway etc. Should use the `'small'` size to avoid unnecessary clutter.
+   *
+   * Note: Read the stepper guidelines for more info.
+   */
+  @property({ type: String })
+  stepperType = 'procedure';
+
+  /** Wheter the stepper is in vertical type. */
+  @property({ type: Boolean })
+  vertical = false;
+
+  /** Stepper size `'large'` & `'small'`. */
+  @property({ type: String })
+  stepperSize = 'large';
+
+  /** Curent index of stepper. Usefull for navigation logic like next, prev etc.*/
+  @property({ type: Number })
+  currentIndex = 0;
```

Comment:
Are we still using this prop? I notice changing it in Storybook does nothing now.

Reply:
I followed carbon and left this for external navigation (like next and prev etc.) as you can see the status stepper story, when user change the index, state can be managed accordingly.